src/impl/helpers.rs

```diff
@@ -31,13 +34,13 @@ pub fn max_align<T>() -> usize {
   core::cmp::max(align_t, header_align)
 }
 
-pub fn make_layout<T>(capacity: usize, alignment: usize) -> alloc::alloc::Layout {
+pub fn make_layout(capacity: usize, alignment: usize, type_size: usize) -> alloc::alloc::Layout {
```

Comment:
It's not clear to me what we gain by removing the generic parameter here.

Reply:
It removes number of instantiations for function, reducing compile times.
If you're curious std library does the same https://github.com/rust-lang/rust/blob/207c80f105282245d93024c95ac408c622f70114/library/alloc/src/raw_vec.rs#L442-L445

By extracting as much code out of generics, you improve compile times automatically equivalent to number of possible T in your code

Reply:
Hmm.

Alright, we can explore this later. 

This PR should focus solely on adding the two new associated functions and the tests.

If we want to work on reducing code bloat, we should first get some metrics set up and then we'll know if our refactors are actually doing anything or not.